CMakeLists.txt

```diff
@@ -42,20 +42,50 @@ set(INC_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include" CACHE PATH "Installation d
 set(MAN_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/share/man" CACHE PATH "Installation directory for manual pages")
 set(PKGCONFIG_INSTALL_DIR "${LIB_INSTALL_DIR}/pkgconfig" CACHE PATH "Installation directory for pkgconfig (.pc) files")
 
-include(CheckTypeSize)
-include(CheckSymbolExists)
-include(CheckFunctionExists)
-include(CheckIncludeFile)
+include(CMakeDependentOption)
+include(CMakePushCheckState)
+include(CheckCCompilerFlag)
 include(CheckCSourceCompiles)
 include(CheckCSourceRuns)
-include(CheckCCompilerFlag)
-include(CMakeDependentOption)
+include(CheckFunctionExists)
+include(CheckIncludeFile)
+include(CheckSymbolExists)
+include(CheckTypeSize)
```

Comment:
Why reorder these? This makes it more difficult to review, since we cannot easily spot any actual changes.
CMakePushCheckState certainly looks interesting though.

Just to be clear, we prefer to have each commit do one thing  (We should put this in the wiki, I guess).
So code-moving/reordering/cleanup in a separate commit/PR to make reviews/audits easier.
Changes like CMakePushCheckState is the kind of thing I'd like to se a separate PR for, and that it changes the old code to use it as well, to keep an common code style.

The more involved the changes of this PR, the less likely I can merge it into the mostly-fixes 2.0 branch. It can still make the 2.1 branch of course, but that does not have an estimated release-date yet.

My preference would be this: (I can still be persuaded otherwise though)
PR1. A minimal fix that enables building on FreeBSD, but does not touch other platforms.
PR2. Rewrite cmake to use CMakePushCheckState.
PR3. Rewrite aligned malloc detection/selection (Effectively this PR).

Reply:
I just ordered them in alphabetical order. This way, each `include()` has only one possible place. I can revert rhis and simply add `include(CMakePushCheckState)` at the end.